src/extension/NUnit2FrameworkDriver.cs

```diff
@@ -66,16 +64,37 @@ public NUnit2FrameworkDriver(AppDomain testDomain)
 
             var initializer = DomainInitializer.CreateInstance(_testDomain);
             initializer.InitializeDomain((int)InternalTrace.Level);
-
-            _runner = RemoteTestRunner.CreateInstance(_testDomain, 1);
         }
 
         public string ID { get; set; }
 
+        private TestRunner Runner {
+            get
+            {
+                if (_runner == null)
+                {
+                    int runnerId;
+                    if (ID == null)
+                    {
+                        throw new NUnitEngineException("NUnit 2 Driver requires that ID must be defined at first.");
+                    }
```

Comment:
TestPackages always have an ID. It is set internally in the constructor and is always the string representation of an int.
DirectTestRunner, which is the only runner that deals with a driver, always sets the driver id to the package id. So both checks are actually redundant.

I have no problem with doing the checks as guard clauses and throwing ArgumentNullException or ArgumentException but we shouldn't throw an NUnitEngineException here. Any error is simply a programming error on our part.